File: pdoo/ruby/test_p1.rb
#encoding:utf-8

require_relative "./combatresult.rb"
require_relative "./dice.rb"
require_relative "./gamecharacter.rb"
require_relative "./loot.rb"
require_relative "./shieldbooster.rb"
require_relative "./shotresult.rb"
require_relative "./suppliespackage.rb"
require_relative "./weapon.rb"
require_relative "./weapontype.rb"

module Deepspace
  class TestP1
    def self.main
      puts "Clase 'Loot':"
      l1 = Loot.new(1,2,3,4,5)
      l2 = Loot.new(5,4,3,2,1)
      puts "El botín 1 tiene: #{l1.nSupplies} #{l1.nWeapons} #{l1.nShields} #{l1.nHangars} #{l1.nMedals}"
      
      puts "Clase 'SuppliesPackage':"
      sp1 = SuppliesPackage.new(1.9,2.8,3.7)
      sp2 = SuppliesPackage.newCopy(sp1)
      puts "El paquete 2 tiene: #{sp2.ammoPower} #{sp2.fuelUnits} #{sp2.shieldPower}"

      puts "Clase 'ShieldBooster':"
      sb1 = ShieldBooster.new("Escudo del poder maximo de la destruccion de los planetas",1.2,3)
      sb2 = ShieldBooster.newCopy(sb1)
      puts "#{sb1.name} tiene como valores: #{sb1.boost} #{sb1.uses}"
      for i in 0..5
        puts "Usando el escudo #{sb2.name}: #{sb2.useIt}"
      end

      puts "Clase 'Weapon':\n"
      w1 = Weapon.new("Espatula",WeaponType::PLASMA,3)
      w2 = Weapon.new("Cuchara",WeaponType::LASER,2)
      puts "Arma 1: poder = #{w1.power}"
      for i in 0..5
        puts "Usando el arma 2: #{w2.useIt}"
      end

      puts "Clase 'Dice':"
      d = Dice.new
      aux1 = 0
      aux2 = 0
      for i in 0..99
        aux1 = aux1 + d.initWithNHangars
        aux2 = aux2 + d.initWithNShields
      end
      puts "Se ha comenzado sin hangares #{100-aux1}/100"
      puts "Se ha comenzado sin escudos #{100-aux2}/100"
      puts "Se ha comenzado con #{d.initWithNWeapons} armas"
      puts "Empieza el jugador #{d.whoStarts(5)} de un total de 5 jugadores"
      puts "Dispara primero #{d.firstShot}"
      if d.spaceStationMoves(0.55)
        puts "La nave con una velocidad 0.55 se mueve"
      else
        puts "La nave con una velocidad 0.55 no se mueve"
      end
    end
  end
end

Deepspace::TestP1.main

File: pdoo/java/Deepspace/TestP1.java
/**
 * @author Manuel Gachs Ballegeer
 */
package Deepspace;

/**
 * @brief Test de la práctica 1
 */
public class TestP1 {/*
    public static void main(String args[]) {
        // Test de la clase Loot
        System.out.print("\n-----Test de la clase Loot-----\n");
        Loot botin = new Loot(1,2,3,4,5);
        System.out.format("Número de suministros:  %d\n",botin.getNSupplies());
        System.out.format("Número de armas:  %d\n",botin.getNWeapons());
        System.out.format("Número de escudos:  %d\n",botin.getNShields());
        System.out.format("Número de hangares:  %d\n",botin.getNHangars());
        System.out.format("Número de medallas:  %d\n",botin.getNMedals());

        // Test de la clase SuppliesPackage
        System.out.print("\n-----Test de la clase SuppliesPackage-----\n");
        SuppliesPackage suministros = new SuppliesPackage(1.5f,3.4f,2.2f);
        SuppliesPackage suministros2 = new SuppliesPackage(suministros);
        System.out.format("Poder de disparo:  %f\n",suministros2.getAmmoPower());
        System.out.format("Cantidad de combustible:  %f\n",suministros2.getFuelUnits());
        System.out.format("Poder del escudo:  %f\n",suministros2.getShieldPower());
        
        // Test de la clase ShieldBooster
        System.out.print("\n-----Test de la clase ShieldBooster-----\n");
        ShieldBooster escudo = new ShieldBooster("Escudo",3.0f,2);
        ShieldBooster escudo2 = new ShieldBooster(escudo);
        System.out.format("Poder del escudo:  %f\n",escudo2.getBoost());
        System.out.format("Número de usos:  %d\n",escudo2.getUses());
        for (int i=1;i<4;i++) {
            float aux = escudo.useIt();
            System.out.format("Uso número %d: ", i);
            if (aux==1.0f)
                System.out.print("No quedan usos\n");
            else
                System.out.print("Escudo usado\n");
        }
        
        // Test de la clase Weapon
        System.out.print("\n-----Test de la clase Weapon-----\n");
        Weapon arma = new Weapon("Arma",WeaponType.LASER,1);
        Weapon arma2 = new Weapon(arma);
        System.out.format("Tipo de arma:  %s\n",arma2.getType());
        System.out.format("Poder del arma %f\n",arma2.power());
        System.out.format("Número de usos:  %d\n",arma2.getUses());
        for (int i=1;i<3;i++) {
            float aux = arma2.useIt();
            System.out.format("Uso número %d: ", i);
            if (aux==1.0f)
                System.out.print("No quedan usos\n");
            else
                System.out.print("Arma usada\n");
        }
        
        // Test de la clase Dice
        System.out.print("\n-----Test de la clase Dice-----\n");
        Dice dado = new Dice();
        float hangares = 0;
        float escudos = 0;
        for(int i = 0; i < 100; i++){
            hangares += dado.initWithNHangars();
            escudos += dado.initWithNShields();
        }
        System.out.format("Probabilidad de no hangares: %f\n", (100-hangares)/100);
        System.out.format("Probabilidad de no escudos: %f\n", (100-escudos)/100);
        System.out.format("Comienzo con %d armas.\n", dado.initWithNWeapons());
        System.out.format("Empieza el jugador: %d\n", dado.whoStarts(5));
        System.out.format("Primer disparo: %s\n", dado.firstShot());
        System.out.format("¿Se ha movido la nave con velocidad 0.55?: %b\n", dado.spaceStationMoves(0.55f));
    }
*/}
